mbed-edge-core-devmode/deb/debian/files/mbed_cloud_client_user_config.h

```diff
@@ -0,0 +1,48 @@
+/*
+ * ----------------------------------------------------------------------------
+ * Copyright 2018 ARM Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ----------------------------------------------------------------------------
+ */
+
+/*
+ * Minimal configuration for using mbed-cloud-client
+ */
+
+#ifndef MBED_CLOUD_CLIENT_USER_CONFIG_H
+#define MBED_CLOUD_CLIENT_USER_CONFIG_H
+
+#define MBED_CLOUD_CLIENT_SUPPORT_CLOUD
+#define MBED_CLOUD_CLIENT_ENDPOINT_TYPE          "MBED_GW"
+#define MBED_CLOUD_CLIENT_TRANSPORT_MODE_TCP
+#define MBED_CLOUD_CLIENT_LIFETIME               3600
+
+#define SN_COAP_MAX_BLOCKWISE_PAYLOAD_SIZE       1024
+#define SN_COAP_DUPLICATION_MAX_MSGS_COUNT       0
+#define SN_COAP_DISABLE_RESENDINGS               1
+
+/* set download buffer size in bytes (min. 1024 bytes) */
+#define MBED_CLOUD_CLIENT_UPDATE_BUFFER          (2 * 1024 * 1024)
+
+/* set the TCP KEEPALIVE values */
+#define MBED_CLIENT_TCP_KEEPALIVE_INTERVAL 60
+#define MBED_CLIENT_TCP_KEEPALIVE_TIME 60
+
+#define PLATFORM_VERSION_HASH_FILE "/var/lib/pelion/platform_version"
+#define PLATFORM_VERSION_FILE "/var/lib/pelion/readable_version"
+
```

Comment:
Should this be in `/etc/` instead of `/var/`? Is it more of a [config thing](https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch03s07.html) or [data that the program itself modifies during](https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch05s08.html#purpose38) runtime?

Reply:
No opinion there.  What is the path you prefer?  `/etc/pelion/platform_version` or just `/etc/platform_version`

Reply:
`/etc/pelion/readable_version` for the human settable thing... and maybe `/var/lib/pelion/platform_version` for the auto hash designed to be generated by the software itself?

Reply:
I've pondered on this for a while because the choice doesn't seem perfectly clear. If we take a look at the most recent [FHS spec](https://refspecs.linuxfoundation.org/FHS_3.0/fhs-3.0.html), there are a few possible candidates for `platform_version` (with `/pelion` appended to each one):
* `/run` ("_system information data describing the system since it was booted_"), with its older variant of `/var/run`. This would be the most fitting definition for version files, but with the caveat that these are transient files ("_files under this directory must be cleared at the beginning of the boot process_") so we'd need to persist our human settable version elsewhere too.
* `/var/lib` ("_state information pertaining to an application or the system_"). This seems like a good choice, until you read the definition for state information ("_data that programs modify while they run_") and the associated restriction which clarifies the purpose as _internal_ state, not general state visible to the user ("_the specific file hierarchy used to store the data must not be exposed to regular users_").
* `/etc` ("configuration files which control the operation of a program"). While the definition doesn't exactly fit the purpose of the version files, an argument can be made that it would be a similar case to `/etc/lsb_release`.

As a last argument to consider, the simplicity of use is important. For this reason, I think both version files should be in the *same path* (whichever it ends up being), as they both represent different angles on similar information.

Reply:
>  so we'd need to persist our human settable version elsewhere too.

I think `/run` is out for that reason

> and the associated restriction which clarifies the purpose as internal state, not general state visible to the user

This makes `/var` suboptimal for `readable_version` IMO

> an argument can be made that it would be a similar case to /etc/lsb_release.

Yeah, that was my though re `readable_version` also.

> As a last argument to consider, the simplicity of use is important.

Makes sense.  I vote both are in `etc` or both are in `/var/lib` then.

Reply:
I just reviewed the snap code, and it seems I got this PR wrong.  According to the implementation on snap, both the [platform_version](https://github.com/PelionIoT/snap-pelion-edge/blob/df8cee937a3e930d1684df8fc2f1a5fde4f3b583/files/edge-core/scripts/launch-edge-core.sh#L57) and the [readable_version](https://github.com/PelionIoT/snap-pelion-edge/blob/df8cee937a3e930d1684df8fc2f1a5fde4f3b583/files/edge-core/scripts/launch-edge-core.sh#L61) are supposed to be auto-generated at startup, and this PR is missing the functionality to populate those files from a user-supplied version_map.json.

Going by that logic, `/var/lib/pelion` makes more sense for those 2 files, and then `/etc/pelion/` could hold the user-supplied version_map.json.

Reply:
If the two files are to be generated each startup, `run/pelion` seems like a perfect fit for them...

Reply:
After some internal discussion, we're going to reduce the scope of the feature and require only the "readable" version file.  To that end, I'd like to change the path to `/etc/pelion/` for both files.  Just to clarify, I do want to keep both files so that the mbed-cloud-client patch file is exactly the same as the snap build to lessen our maintenance burden, but we just won't have any scripts that auto-populate the file.  So the commit message will stay the same and both files will be read on startup, and I'll just change the path to `/etc/pelion/`.  Does that sound ok?